### PR TITLE
changed ports as per issue #3

### PR DESCRIPTION
**Pull request by hkiiita**
Hi @coderbyheart request you to please review the changes for issue #3 and please comment on any further modifications. Thank you.

### CODE REVIEW

**Review by coderbyheart — CHANGES_REQUESTED** [review]
Looks good (trivial change).
However to not break user experience we need to keep the service also exposed on the previously used ports.

**Comment by hkiiita** [comment]
Thanks for replying @coderbyheart ; learnt this new thing today that we indeed need to keep the service alive on old ports for not breaking end user experience. Will work on it and get back to you for more learning and guidance. Thanks again.

**Comment by hkiiita** [comment]
Hi @coderbyheart , please review . Besides the changes made, i did not add old ports in the client.go file as , i guess, we want new clients to access the server on new ports, although, clients with old ports may still continue to access the service.

I have already learnt a thing or two from this work. Looking forward to your further guidance and mentoring on this. Thanks again !

**Review by coderbyheart — CHANGES_REQUESTED** [review]
Tests needed that demonstrate that old ports are still used. Client.go can be modified since it's only used for testing.

**Comment by hkiiita** [comment]
> Tests needed that demonstrate that old ports are still used. Client.go can be modified since it's only used for testing.

Hi @coderbyheart , thanks for being with me all along. Have added the tests for both old and new ports, please check and review and let me know for modifications / improvements.

Thanks !

File: server/server.go [hunk 1]
```diff
@@ -20,6 +20,20 @@ func check(e error) {
 	}
 }
 
+type serverConfig struct {
+	oldUdpPort  int
```
Comment (coderbyheart) [inline server/server.go]: Use same order of words in keys here: old/new<protocol type>

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: server/server.go (hunk 1 above)
Comment (hkiiita) [reply server/server.go]: Hi , 

existing order is as 
```go
type serverConfig struct {
	oldUdpPort  int
	newUdpPort  int
	dtlsOldPort int
	dtlsNewPort int
}
```

instead, did you mean the following way ? 

```go
type serverConfig struct {
	oldUdpPort  int
	newUdpPort  int
	oldDtlsPort int
	newDtlsPort int
}
```

Comment (coderbyheart) [reply server/server.go]: Yes.

Comment (hkiiita) [reply server/server.go]: Hi @coderbyheart , please review the changes. Thanks.

File: server/server.go [hunk 2]
```diff
@@ -46,10 +60,29 @@ func main() {
 
 	r := internal.NewServer(storageClient, containerName)
 
-	udpPort := 5688
-	dTLSPort := 5689
-	udpAddr := fmt.Sprintf("%s:%d", *address, udpPort)
-	dtlsAddr := fmt.Sprintf("%s:%d", *address, dTLSPort)
+	go func() {
```
Comment (coderbyheart) [inline server/server.go]: Refactor this to reuse the code that launches the server and not duplicate it.

Comment (hkiiita) [reply server/server.go]: HI @coderbyheart , thanks for being with me. That indeed was another learning point for me towards maintaining clean code. Request some of your busy time to please review this change. Thanks again !!

**Comment by coderbyheart** [comment]
Tests pass: https://github.com/NordicPlayground/nordic-developer-academy-coap/actions/runs/9125300745/job/25091193741

File: client/client.go [hunk 3]
```diff
@@ -34,25 +38,40 @@ func main() {
...
+func testPorts(flagValues flags, udpPort int, dTLSPort int) {
+
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(10)*time.Second)
 	defer cancel()
 
-	udpPort := 5688
-	dTLSPort := 5689
 	var udpAddr string
 	var dtlsAddr string
+	_ = dtlsAddr
```
Comment (coderbyheart) [inline client/client.go]: Why is this needed?

File: server/server.go [hunk 4]
```diff
@@ -46,24 +75,31 @@ func main() {
...
-	dtlsAddr := fmt.Sprintf("%s:%d", *address, dTLSPort)
+	// old ports
+	go launchServer(flagValues, ports.oldUdpPort, ports.oldDtlsPort, r)
+
+	// new ports
+	launchServer(flagValues, ports.newUdpPort, ports.newDtlsPort, r)
+
+}
+
+func launchServer(flagValues flags, udpPort int, dtlsPort int, r *mux.Router) {
```
Comment (coderbyheart) [inline server/server.go]: Move to new module

File: client/client.go (hunk 3 above)
Comment (hkiiita) [reply client/client.go]: yeah, it crept in while testing locally i guess. Removed it. :-)

File: server/server.go (hunk 4 above)
Comment (hkiiita) [reply server/server.go]: Thanks for the guidance, yes, moved it like this  --> [7938d5c](https://github.com/NordicPlayground/nordic-developer-academy-coap/pull/4/commits/7938d5c15d0677b445c8282a8681c343a7505864)

**Review by coderbyheart — APPROVED** [review]
